atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -2378,6 +2364,37 @@ private boolean rollbackOtherTransaction(long startTs, TransactionService transa
         }
     }
 
+    ///////////////////////////////////////////////////////////////////////////
+    /// Post-commit actions
+    ///////////////////////////////////////////////////////////////////////////
+
+    private void runPostCommitActions() {
+        // Run close() to release locks before running success callbacks, since success callbacks might
+        // start a new transaction and attempt to grab the same locks as the current transaction.
+        close();
+
+        runSuccessCallbacksIfDefinitivelyCommitted();
+    }
+
+    private void close() {
+        try {
+            State finalState = state.get();
+            checkState(
```

Comment:
If this throws, you won't run the closer? Won't that leak?

Reply:
Removed this check - think it's fine to always run closer.